x-pack/plugins/cases/public/components/all_cases/columns_popover.test.tsx

```diff
@@ -13,173 +13,180 @@ import type { AppMockRenderer } from '../../common/mock';
 import { createAppMockRenderer } from '../../common/mock';
 import { ColumnsPopover } from './columns_popover';
 
-describe('ColumnsPopover', () => {
-  let appMockRenderer: AppMockRenderer;
+for (let i = 0; i < 500; i++) {
+  describe('ColumnsPopover', () => {
+    let appMockRenderer: AppMockRenderer;
 
-  beforeEach(() => {
-    appMockRenderer = createAppMockRenderer();
-    jest.clearAllMocks();
-  });
+    beforeEach(() => {
+      appMockRenderer = createAppMockRenderer();
+      jest.clearAllMocks();
+    });
 
-  const selectedColumns = [
-    { field: 'title', name: 'Title', isChecked: true },
-    { field: 'category', name: 'Category', isChecked: false },
-    { field: 'tags', name: 'Tags', isChecked: false },
-  ];
+    const selectedColumns = [
+      { field: 'title', name: 'Title', isChecked: true },
+      { field: 'category', name: 'Category', isChecked: false },
+      { field: 'tags', name: 'Tags', isChecked: false },
+    ];
 
-  it('renders correctly a list of selected columns', async () => {
-    appMockRenderer.render(
-      <ColumnsPopover selectedColumns={selectedColumns} onSelectedColumnsChange={() => {}} />
-    );
+    it('renders correctly a list of selected columns', async () => {
+      appMockRenderer.render(
+        <ColumnsPopover selectedColumns={selectedColumns} onSelectedColumnsChange={() => {}} />
+      );
 
-    userEvent.click(await screen.findByTestId('column-selection-popover-button'));
+      userEvent.click(await screen.findByTestId('column-selection-popover-button'));
 
-    expect(await screen.findByTestId('column-selection-popover')).toBeInTheDocument();
+      expect(await screen.findByTestId('column-selection-popover')).toBeInTheDocument();
+      expect(
+        await screen.findByTestId(`column-selection-switch-${selectedColumns[0].field}`)
```

Comment:
Checked first selectedColumn with `screen.findByTestId` so that it waits for columns element to render, and then looped through the rest.

Reply:
how about adding `await waitForEuiPopoverOpen();` after the clicks on `'column-selection-popover-button'`?

Reply:
yeah.. good idea 👍